src/nvim/api/vim.c

```diff
@@ -837,6 +837,85 @@ Array nvim_get_api_info(uint64_t channel_id)
+///                 possible keys (mostly optional):
+///     - "major" major version (required, set to 0 if no release yet)
+///     - "minor"
+///     - "patch"
+///     - "prerelease"
+///     - "commit"
+/// @param type Must be one of the following values
+///     - "remote" remote client that connected to nvim
+///     - "ui" gui frontend
+///     - "host" plugin host, typically started by nvim
+///     - "plugin" single plugin, started by nvim
+/// @param methods Builtin methods in the client. For a host, this does not
+///                include plugin methods which will be discovered later.
+///                The key should be the method name, the values are dicts with
+///                the following (optional) keys:
+///     - "async"  if true, send as a notification. If false or unspecified,
+///                use a blocking request
+///     - "n_args" Number of arguments. Could be a single integer or an array
+///                two integers, minimum and maximum inclusive.
+///     Futher keys might be added in later versions of nvim and unknown keys
+///     are thus ignored. Clients must only use keys defined in this or later
+///     versions of nvim!
+///
+/// @param attribues Informal attributes describing the client. Clients might
+///                  define their own keys, but the following are suggested:
+///     - "website" Website of client (for instance github repository)
+///     - "license" Informal descripton of the licence, such as "Apache 2",
+///                 "GPLv3" or "MIT"
+///     - "logo"    URI or path to image or SVG logo
+///
+void nvim_set_client_info(uint64_t channel_id, String name,
+                           Dictionary version, String type,
+                           Dictionary methods, Dictionary attributes)
+  FUNC_API_SINCE(4) FUNC_API_REMOTE_ONLY
+{
+  Dictionary info = ARRAY_DICT_INIT;
+  PUT(info, "name", copy_object(STRING_OBJ(name)));
+  PUT(info, "version", copy_object(DICTIONARY_OBJ(version)));
+  PUT(info, "type", copy_object(STRING_OBJ(type)));
+  PUT(info, "methods", copy_object(DICTIONARY_OBJ(methods)));
+  PUT(info, "attributes", copy_object(DICTIONARY_OBJ(attributes)));
+
+  rpc_set_info(channel_id, info);
+}
+
+/// Get information about all open channels
+///
+/// @returns Array of Dictionaries, each describing a channel with the
+/// following keys:
+///     - "stream"  the stream underlying the channel
+///         - "stdio"      stdin and stdout of this Nvim instance
+///         - "stderr"     stderr of this Nvim instance
+///         - "socket"     TCP/IP socket or named pipe
+///         - "job"        job with communication over its stdio
+///    -  "mode"    how data received on the channel is interpreted
+///         - "bytes"      send and recieve raw bytes
+///         - "terminal"   a |terminal| instance interprets ASCII sequences
+///         - "rpc"        |RPC| communication on the channel is active
+///    -  "pty"     Name of pseudoterminal if present (optional)
```

Comment:
Could this go in a generic"name" field? Seems like a good thing to have anyway.

Reply:
`"pty"` is slightly more precise, if we have a _pty job_, it is not obvious if "name" describes the process or the pty. This fullfills the specific use case of needing to know the device name of the pty, Maybe should mention `/dev/pts/xx` explicitly so if you want it you know that you have found it. Also the key is present at all (if and) only if a pty is used, and serves as indicator.